pkg/log/io/leveler.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io
+
+import (
+	"fmt"
+
+	"log/slog"
+)
+
+// slogLeveler is the leveler for the logger
+type slogLeveler struct {
+	level slog.Level
```

Comment:
is this logic used? I dont understand the porpuse of it, can you explain it?

Reply:
The default slogLevel is inmutable and can't be changed. We need to initialise the logger and change it depending on the output format of several commands. This struct allows to change the level on the fly without the need of defining a whole new logger, which makes it more efficient and reusable